spec/models/book_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe Book, type: :model do
+  it 'is not valid without edition' do
+    expect(FactoryGirl.build(:book, edition: nil)).not_to be_valid
+  end
+
+  it 'is not valid without title' do
+    expect(FactoryGirl.build(:book, title: nil)).not_to be_valid
+  end
+
+  it 'is not valid when edition is non-numeric' do
+    expect(FactoryGirl.build(:book, edition: 't')).not_to be_valid
+  end
+
+  it 'is valid with title and edition numeric' do
+    expect(FactoryGirl.create(:book)).to be_valid
+  end
+
+  describe '#name_with_initial' do
+    let(:book) { FactoryGirl.create(:book, title: 'Fantasy Book', edition: 2) }
```

Comment:
You don't need to save book to the db to check the `name_with_initial`.
It's enough to use `.build` instead of `.create` here.
This is generic rule for tests: don't save the data to the db if you don't have to. It speeds up the tests.